### ISSUE TITLE
Timer nodes with wait times lower than 1 are not in sync with fps

### ISSUE THREAD

**Issue by Nukiloco** [godotengine/godot#52687]
### Godot version

v3.4.beta4.official [6a058cbf3]

### System information

Windows 10

### Issue description

When setting the wait_timer to values around 0.01, the amount of timeout callbacks are different between a game running at 144 FPS and a game running at 5 FPS.

I expect the timer to be synced up with the fps or delta with wait_times on Timer nodes around 0.01.

### Steps to reproduce

1. Set the debug fps set to 0.
2. Run the project.
3. Wait around 5 seconds.
4. Notice how the amount is around 500.
5. Set the debug fps to 5.
6. Run the project.
7. Wait around 5 seconds.
8. Notice how the amount is around 40.

### Minimal reproduction project

[TimerNodeIssue.zip](https://github.com/godotengine/godot/files/7166921/TimerNodeIssue.zip)

**Comment by Ev1lbl0w** [comment]
The timer value is processed on the engine loop and it triggers only once per frame, so with lower FPS the accuracy will degrade. I'll open a PR to propose having it triggering multiple times in a frame for your scenario.

**Comment by Nukiloco** [comment]
> 
> 
> The timer value is processed on the engine loop and it triggers only once per frame, so with lower FPS the accuracy will degrade. I'll open a PR to propose having it triggering multiple times in a frame for your scenario.

Thanks! And yea this is something I thought that was possibly happening in the engine side.

**Comment by Calinou** [comment]
Closing in favor of https://github.com/godotengine/godot-proposals/issues/3386, as changing this behavior has significant implications that need to be discussed further.
